src/testswarm-browserstack.js

```diff
@@ -407,6 +409,18 @@ self = {
 					precision += pts;
 				} );
 
+				// Handle the "os" field in a special way - some browsers on BrowserStack only
+				// define the major part and we want to be able to prefer them when there's
+				// another entry with a matching major that also specifies the minor.
+				// One example is iOS: BrowserStack includes both "10" & "10.3"; the former
+				// is a real device and the latter is an emulator; we want the real device.
+				if ( !tsUaSpec.osMinor && !bswUaSpec.osMinor ) {
```

Comment:
It took me a while to realise that this doesn't overlap with the `each` above. The reason is that both are conditional on `tsUaSpec` not having the given property, so this is basically a fallback way that still adds some precision points even if we didn't have `os*` specified.

Makes sense now. I'll land with an added comment.